kpack-image-builder/controllers/buildreconcilerinfo_controller.go

```diff
@@ -0,0 +1,170 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	buildv1alpha2 "github.com/pivotal/kpack/pkg/apis/build/v1alpha2"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"code.cloudfoundry.org/korifi/controllers/api/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	BuildReconcilerInfoName = "kpack-image-builder"
+	ReadyConditionType      = "Ready"
+)
+
+func NewBuildReconcilerInfoReconciler(c client.Client, scheme *runtime.Scheme, log logr.Logger, clusterBuilderName, rootNamespaceName string) *BuildReconcilerInfoReconciler {
+	return &BuildReconcilerInfoReconciler{
+		Client:             c,
+		Scheme:             scheme,
+		Log:                log,
+		ClusterBuilderName: clusterBuilderName,
+		RootNamespaceName:  rootNamespaceName,
+	}
+}
+
+// BuildReconcilerInfoReconciler reconciles a BuildReconcilerInfo object
+type BuildReconcilerInfoReconciler struct {
+	client.Client
+	Scheme             *runtime.Scheme
+	Log                logr.Logger
+	ClusterBuilderName string
+	RootNamespaceName  string
+}
+
+//+kubebuilder:rbac:groups=korifi.cloudfoundry.org,resources=buildreconcilerinfos,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=korifi.cloudfoundry.org,resources=buildreconcilerinfos/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=korifi.cloudfoundry.org,resources=buildreconcilerinfos/finalizers,verbs=update
+
+//+kubebuilder:rbac:groups=kpack.io,resources=clusterbuilders,verbs=get;list;watch
+//+kubebuilder:rbac:groups=kpack.io,resources=clusterbuilders/status,verbs=get
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the BuildReconcilerInfo object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
+func (r *BuildReconcilerInfoReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	if req.Name != BuildReconcilerInfoName || req.Namespace != r.RootNamespaceName {
+		return ctrl.Result{}, nil
+	}
+
+	info := new(v1alpha1.BuildReconcilerInfo)
+	err := r.Client.Get(ctx, req.NamespacedName, info)
+	if err != nil {
+		r.Log.Error(err, "Error when fetching BuildReconcilerInfo") // TODO: ensure we're using correct log format
```

Comment:
Are these logging TODOs still valid?

Reply:
Thanks for the reminder. I'm going to take them out and we can address logging inconsistencies all at once later.